engine/src/execution_graph/logic_controllers/BatchProcessing.cpp

```diff
@@ -266,6 +272,11 @@ kstatus BindableTableScan::run() {
         std::iota(projections.begin(), projections.end(), 0);
     }
 
+    if (provider->is_sql()) {
+      auto sql_provider = std::dynamic_pointer_cast<ral::io::abstractsql_data_provider>(provider);
```

Comment:
this logic of setting the column indices that you want should be handled by the CacheDataIO. all the other data types handle it there.

Reply:
Done, now CacheDataIO handle that part